com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Utility/Imposter.hlsl

```diff
@@ -0,0 +1,393 @@
+
+    float2 gridFrac = frac(grid);
+    float2 gridFloor = floor(grid);
+
+    //convert frame coord to octahedron ray
+    float3 frame0ray = normalize(GridToVector(float2(gridFloor / framesMinusOne * 2 - 1), isHemi));
+
+    float3 center = float3(0, 0, 0);
+
+    float3 plane0x;
+    float3 plane0normal = frame0ray;
+    float3 plane0z;
+    float3 frame0local = FrameTransformLocal(BillboardPosToCam, frame0ray, plane0x, plane0z);
+
+    float2 vUv0 = PlaneIntersectionUV(plane0normal, plane0x, plane0z, center, size, ray);
+    vUv0 /= imposterFrames;
+
+    frame0local.xy /= imposterFrames;
+
+    float2 gridSnap = floor(grid) / imposterFrames.xx;
+    float2 frame0grid = gridSnap;
+    float2 vp0uv = frame0grid + vUv0.xy;
+
+    if (parallax != 0) {
+        float h0 = SAMPLE_TEXTURE2D(heightMap, ss, vp0uv)[heightMapChannel];
+        vp0uv += (0.5 - h0) * frame0local.xy * parallax;
+    }
+
+    //vert pos
+    outPos = BillboardPos + imposterOffset;
+
+    //surface
+    outUV0 = vp0uv;
+    outUVGrid.xy = UVscaled;
+    outUVGrid.zw = grid;
+
+}
+//Sample from UVs 
+void ImposterSample(in float imposterFrames, in texture2D Texture, in float4 mapTexelSize, in float4 weights, in float imposterFrameClip, in float4 inUVGrid,
+    in float2 inUV0, in float2 inUV1, in float2 inUV2, in SamplerState ss, out float4 outColor)
+{
+    //frame size ->2048/12 = 170.6
+    float frameSize = mapTexelSize.z / imposterFrames;
+    //actual texture size used -> 170*12 = 2040
+    float actualTextureSize = floor(frameSize) * imposterFrames;
+    //the  scalar -> 2048/2040 = 0.99609375
+    float scalar = mapTexelSize.z / actualTextureSize;
+
+    inUV0 *= scalar;
+    inUV1 *= scalar;
+    inUV2 *= scalar;
+
+    float2 gridSnap = floor(inUVGrid.zw) / imposterFrames.xx;
+    float2 frame0grid = gridSnap;
+    float2 frame1grid = gridSnap + (lerp(float2(0, 1), float2(1, 0), weights.w) / imposterFrames.xx);
+    float2 frame2grid = gridSnap + (float2(1, 1) / imposterFrames.xx);
+
+    //clip out neighboring frames 
+    float2 gridSize = 1.0 / imposterFrames.xx;
+    float2 bleeds = mapTexelSize.xy * imposterFrameClip;
+    inUV0 = clamp(inUV0, frame0grid - bleeds, frame0grid + gridSize + bleeds);
+    inUV1 = clamp(inUV1, frame1grid - bleeds, frame1grid + gridSize + bleeds);
+    inUV2 = clamp(inUV2, frame2grid - bleeds, frame2grid + gridSize + bleeds);
+
+    outColor = SAMPLE_TEXTURE2D_GRAD(Texture, ss, inUV0, ddx(inUVGrid.xy), ddy(inUVGrid.xy)) * weights.x
+        + SAMPLE_TEXTURE2D_GRAD(Texture, ss, inUV1, ddx(inUVGrid.xy), ddy(inUVGrid.xy)) * weights.y
+        + SAMPLE_TEXTURE2D_GRAD(Texture, ss, inUV2, ddx(inUVGrid.xy), ddy(inUVGrid.xy)) * weights.z;
+}
+
+void ImposterSample_oneFrame(in float imposterFrames, in texture2D Texture, in float4 mapTexelSize, in float imposterFrameClip, in float4 inUVGrid,
+    in float2 inUV0, in SamplerState ss, out float4 outColor)
+{
+    //frame size ->2048/12 = 170.6
```

Comment:
Can lines 373-388 be done in the ImposterUV_oneFrame function instead? If so, we could get rid of the Frames input on this node.